spock-core/src/main/java/org/spockframework/mock/runtime/MockController.java

```diff
@@ -38,31 +41,31 @@ public MockController() {
 
   @Override
   public Object handle(IMockInvocation invocation) {
-    Optional<IMockInteraction> interaction = findInteraction(invocation);
-    if (interaction.isPresent()) {
-      try {
-        return interaction.get().accept(invocation);
-      } catch (InteractionNotSatisfiedError e) {
-        synchronized (this) {
-          errors.add(e);
+    Supplier<Object> resultSupplier = null;
+    synchronized (this) {
```

Comment:
Why do you use a dedicated `LOCK` object in the other classes, but `this` here?

Reply:
Is it ok or do we need to change something?